server/cloudapps/on_user_joined_channel.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package cloudapps
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+type UserJoinedChannelNotification struct {
```

Comment:
The [Mattermost Cloud Apps](https://docs.google.com/document/d/1r52PPwVfcyCjOgnq5ulmrLdeOVgDSGuTbhqu9Cjq9GE/edit#heading=h.qksj0bkjfz31) describes these as Notification Subjects.  The list of subjects is a considerable number with the potential to grow.  I am wondering if we'd benefit from creating another package (subjects?) inside `cloudapps`?

Reply:
maybe at some point it maybe worth moving the entire notification proxy into its own package, but 0/5 no reason to do it just yet.

Reply:
sure. I'm also 0/5 and was just throwing it out there.  Non-blocking nit